native/explorer/src/datatypes.rs

```diff
@@ -407,28 +417,21 @@ pub struct ExDateTime<'a> {
     pub zone_abbr: &'a str,
 }
 
-// impl From<i64> for ExDateTime<'_> {
-//     fn from(microseconds: i64) -> Self {
-//         timestamp_to_datetime_utc(microseconds).into()
-//     }
-// }
-
-impl From<ExDateTime<'_>> for i64 {
-    fn from(dt: ExDateTime<'_>) -> i64 {
-        let duration = NaiveDate::from_ymd_opt(dt.year, dt.month, dt.day)
-            .unwrap()
-            .and_hms_micro_opt(dt.hour, dt.minute, dt.second, dt.microsecond.0)
-            .unwrap()
-            .signed_duration_since(
-                NaiveDate::from_ymd_opt(1970, 1, 1)
-                    .unwrap()
-                    .and_hms_opt(0, 0, 0)
-                    .unwrap(),
-            );
-
-        match duration.num_microseconds() {
-            Some(us) => us,
-            None => duration.num_milliseconds() * 1_000,
+pub fn ex_datetime_to_timestamp_res(
+    ex_datetime: ExDateTime,
+    time_unit: TimeUnit,
+) -> Result<i64, ExplorerError> {
```

Comment:
1. I'm returning a `Result` here so we get better error messages elsewhere.
2. Is adding a `_res` suffix to a function that return `Result` weird? Do people do that?

Reply:
Reverted the `_res` thing: [e9cac12](https://github.com/elixir-explorer/explorer/pull/1020/commits/e9cac12ea984ca7027c7d6e7c505a54dc75f8688)